lbrynet/lbryfilemanager/EncryptedFileManager.py

```diff
@@ -74,12 +75,49 @@ def _add_to_sd_identifier(self):
         self.sd_identifier.add_stream_downloader_factory(
             EncryptedFileStreamType, downloader_factory)
 
-    def _start_lbry_files(self):
+    def _check_stream_is_managed(self, stream_hash):
+        # check that all the streams in the stream_info_manager are also
+        # tracked by lbry_file_manager and fix any streams that aren't.
+        def _check_rowid_result(rowid):
+            if rowid is not None:
+                return defer.succeed(True)
+            raise MissingLBRYFile
+
+        def _fix_missing_file(err):
+            if err.check(MissingLBRYFile):
+                rate = self.session.base_payment_rate_manager.min_blob_data_payment_rate
+                d = self.stream_info_manager.get_stream_info(stream_hash)
+                d.addCallback(lambda info: log.warning("Trying to fix missing lbry file for %s",
+                                                       info[1].decode('hex')))
+                d.addCallback(lambda _: self._save_lbry_file(stream_hash, rate))
+                return d
+
+        d = self._get_rowid_for_stream_hash(stream_hash)
+        d.addCallback(_check_rowid_result)
+        d.addErrback(_fix_missing_file)
+        return d
```

Comment:
this would be nicer as inlineCallbacks

```
@defer.inlineCallbacks
def _check_stream_is_managed(self, stream_hash):
    row_id = yield self._get_rowid_for_stream_hash(stream_hash)
    if rowid:
        defer.returnValue(True)
    else:
        rate = self.session.base_payment_rate_manager.min_blob_data_payment_rate
        info = yield self.stream_info_manager.get_stream_info(stream_hash)
        # as an aside, I think its bad practice to refer to tuples like this
        # and I prefer to unpack the tuple into descriptive names so that the reader doesn't have to look up the 
        # get_stream_info function to see what is returned.  Alternatively, use namedtuples
        # 
        # Also, we shouldn't need to remember how to log each object in our codebase.  Save yourself
        # the effort and put in place good __repr__ and __str__ functions so that we can just log
        #
        log.warning("Trying to fix missing lbry file for %s", info[1].decode('hex')))
        yield self._save_lbry_file(stream_hash, rate)
```